lua/bufferline/state.lua

```diff
@@ -92,8 +91,12 @@ function M.get_buffer_data(id)
   return M.buffers_by_id[id]
 end
 
+local main_module = nil
 function M.update()
-  bufferline.update()
+  if main_module == nil then
+    main_module = require'bufferline'
+  end
+  main_module.update()
```

Comment:
@Iron-E This is going to conflict with your changes in #269. I'm doing this here because there is a cyclical dependency loop otherwise, `bufferline.update()` is needed but also requires part of the state module because it's so central. I don't see an easy fix and I'm a bit too lazy to do a cleaner refactoring, imho we should keep this here but I'm open to suggestions.

Reply:
I ran into the dep cycle while making #269. My solution was to remove `local state = require'bufferline.state'` from the top of the file, and use `require'bufferline.state'.foo` directly inside command callbacks (since they are only evaluated when the command is called, `bufferline.lua` can load without `state.lua` in scope)

Reply:
Ok sounds good, I did it this way for now.